separator_models/porous_separator.py

```diff
@@ -47,12 +47,23 @@ def __init__(self, input_file, inputs, params, offset):
             raise ValueError('Please specify a valid electrolyte transport ',
                 'model.')
 
-        if inputs['transport']['diffusion-scaling'] == 'ideal':
+        try:
+            if inputs['transport']['diffusion-scaling'] == 'ideal':
+                self.scale_diff = transport.scale_diff_ideal
+            elif inputs['transport']['diffusion-scaling'] == 'zhang':
+                self.scale_diff = transport.scale_diff_zhang
+                self.n_Li_atoms = np.zeros(self.elyte_obj.n_species)
+                for i, species in enumerate(self.elyte_obj.species_names):
+                    self.n_Li_atoms[i] = self.elyte_obj.n_atoms(species, 'Li')
+
+                self.C_Li_0 = self.C_k_0[self.index_Li] + \
+                                            np.dot(self.n_Li_atoms, self.C_k_0)
+            else:
```

Comment:
Does it work to have an `else` + `ValueError` inside of a `try` + `except`  handler?  If there is no valid scaling model, shouldn't it just default to ideal?